apis/v1alpha1/clusterprofile_types.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterProfileSpec defines the desired state of ClusterProfile
+type ClusterProfileSpec struct {
+	// DisplayName defines a human-readable name of the ClusterProfile
+	DisplayName string `json:"displayName"`
```

Comment:
optional field?

Reply:
I think it should be required. The consumer of this API might not be able to use the ClusterProfile resource name for display purposes because if the producer follows our guideline then the resource name has be rather unique to avoid clashing. The consumer of this API will probably want to rely on having a displayName to show users the friendly name of the cluster in their UI. If it's optional and sometimes its empty then the consumer can't rely on this field as much.
WDYT?

Reply:
+1 on required

I suggest adding `// +required` tag on each required field and adding `// +optional` on each optional field, to make it clear.

Reply:
based on my understanding on the field name/comment, this field is for human read only. It should be optional and depends on whether users really need this display name or not.